Userland/Libraries/LibCrypto/Hash/SHA1.cpp

```diff
@@ -67,6 +83,188 @@ inline void SHA1::transform(u8 const* data)
     secure_zero(blocks, 16 * sizeof(u32));
 }
 
+#if ARCH(I386) || ARCH(X86_64)
+sha1_attribute_target_x86 static void transform_impl(u32* state, u8 const* data)
+{
+    __m128i reverse_8;
+    __m128i abcd_1;
+    __m128i e;
+    __m128i old_abcd;
+    __m128i old_e;
+    __m128i msg_0;
+    __m128i abcd_2;
+    __m128i msg_1;
+    __m128i msg_2;
+    __m128i msg_3;
+
+    VERIFY(state);
+    VERIFY(data);
+
+    // Set up constant for reversing input buffer.
+    reverse_8 = _mm_set_epi64x(0x0001020304050607ull, 0x08090a0b0c0d0e0full);
+    // Load state into working registers.
+    abcd_1 = _mm_loadu_si128(reinterpret_cast<__m128i const*>(&state[0]));
```

Comment:
Does all of this *really* need to be 100% unrolled? It's very hard to follow, and uses all these temporary variables in a very unintuitve way.

Reply:
Also what are 'abcd' and 'e'? Are they names from the spec? or just arbitrary temp variable names? Descriptive names are much preferred.

Reply:
Yes, need to be unrolled as some of the intrinsics / instructions consume only constant integers, not integers from variables.

Reply:
Yes, the spec, NIST FIPS PUB 180-4 is talking about variables with such names. For example in sections 6.2.2.2 and 6.2.2.3.